utils/clflags.ml

```diff
@@ -567,17 +568,36 @@ let add_arguments loc args =
       arg_names := String.Map.add arg_name loc !arg_names
   ) args
 
-let print_arguments usage =
-  Arg.usage !arg_spec usage
+let create_usage_msg program =
+  Printf.sprintf "Usage: %s <options> <files>\n\
+    Try '%s --help' for more information." program program
+
+
+let print_arguments program =
+  Arg.usage !arg_spec (create_usage_msg program)
+
 
 (* This function is almost the same as [Arg.parse_expand], except
    that [Arg.parse_expand] could not be used because it does not take a
    reference for [arg_spec].*)
-let parse_arguments argv f msg =
+let parse_arguments ?(current=ref 0) argv f program =
   try
-    let argv = ref argv in
-    let current = ref 0 in
-    Arg.parse_and_expand_argv_dynamic current argv arg_spec f msg
+    Arg.parse_and_expand_argv_dynamic current argv arg_spec f "\209"
```

Comment:
I understand the idea of using a "marker" character here, but why `\209`? (Apparently in Unicode this is Ñ, which indeed probably does not occur in error messages.). Wouldn't `\000` make sense as well? In any case, a small comment to explain why we pass a marker here could be nice.

Reply:
Thanks! \000 works too, added a comment to explain it's usage